src/fromager/bootstrapper.py

```diff
@@ -101,19 +104,33 @@ def bootstrap(self, req: Requirement, req_type: RequirementType) -> Version:
         build_env = None
         sdist_root_dir = None
         if not pbi.pre_built:
-            (
-                sdist_root_dir,
-                unpack_dir,
-                build_dependencies,
-            ) = self._prepare_source(req, source_filename, resolved_version)
-
-            found_wheel_filename = self._is_wheel_built(req, resolved_version)
-            if not found_wheel_filename:
-                wheel_filename, build_env = self._build(
-                    req, resolved_version, sdist_root_dir, build_dependencies
+            unpacked_cached_wheel, cached_wheel_filename = (
```

Comment:
This section looks correct, but might actually be clearer if the stages were handled one after the other instead of having 2 branches.

```
if not unpacked_cached_wheel:
  # prepare source

# prepare build dependencies (always, so no "if" statement)

if not unpacked_cached_wheel and not cached_wheel_filename:
  # build
```

That sort of organization shows that it's a linear flow through a set of steps but that we skip some steps. If we add a step later, we can add it to the sequence at the right point with the right conditions.

It would also let us move the logic into a separate function (`_handle_building_wheel`?) that just returns early if it can download the wheel from the cache.